source/zpeep-manager.js

```diff
@@ -1,324 +1,372 @@
-              const requests = [];
-
-              if (projectId) {
-                //Gets the project info:
-                const requestProjectInfo = {
-                  url: `${CONFIG.BASECAMP_PROTOCOL}${CONFIG.BASECAMP_TOKEN}@${CONFIG.BASECAMP_DOMAIN}/projects/${projectId[0]._}.xml`,
-                  headers: {'User-Agent': REQUEST_USER_AGENT_HEADER}
-                };
-                //console.log(requestProjectInfo.url);
-
-                requests.push(requestURL(requestProjectInfo));
-              }
-
-              if (todoItemId && todoItemId[0] && todoItemId[0]._) {
-                //Gets the todo item name
-                const requestTodoInfo = {
-                  url: `${CONFIG.BASECAMP_PROTOCOL}${CONFIG.BASECAMP_TOKEN}@${CONFIG.BASECAMP_DOMAIN}/todo_items/${todoItemId[0]._}.xml`,
-                  headers: {'User-Agent': REQUEST_USER_AGENT_HEADER}
-                };
-                //console.log(requestTodoInfo.url);
-
-                requests.push(requestURL(requestTodoInfo));
-              }
-
-              if (requests.length) {
-                Promise.all(requests)
-                    .spread((responseProjectInfo, responseTodoInfo) => {
-
-                      const projectName = cheerio.load(responseProjectInfo).root().find('project > name').text();
-
-                      report.projectName = projectName;
-
-                      if (typeof responseTodoInfo !== 'undefined') {
-                        const todoName = cheerio.load(responseTodoInfo).root().find('todo-item > content').text();
-
-                        report.todoName = todoName + ': ';
-                      }
-                    }).catch((err) => {
-                      console.log('Error occured when requesting additional entry information:' + err);
-                    }).finally(() => {
-                      currentTimeEntry++;
-                      //All requests made, so calling callback to continue the rendering process
-                      if (currentTimeEntry === timeEntriesCount) {
-                        //Returns array of formalized data
-                        //console.log(currentTimeEntry);
-                        //console.log(timeEntries[20].report[0]);
-                        callback(timeEntries);
-                      }
+    },
+
+    /**
+     * Get timesheet reports and format data for proper rendering
+     * @param  {string}   reportDate A string representing a date 'YYYYMMDD'
+     * @param  {Function} callback   Triggers once data is retrieved
+     * @returns {void}
+     */
+    getZPeepsTimeReport (reportDate, callback) {
+
+        //Initialize Object with time report data
+        let timeEntries = null;
+        const REQUEST_USER_AGENT_HEADER = 'Andres Garcia Reports (andres@zemoga.com)';
+
+        //Call to Basecamp reports
+        const requestTimeReport = {
+            uri: `${CONFIG.BASECAMP_PROTOCOL}${CONFIG.BASECAMP_TOKEN}@${CONFIG.BASECAMP_DOMAIN}${CONFIG.BASECAMP_PATH}`,
```

Comment:
Good catch! interesting... How did you figure that out? I'm checking the docs and it is not mentioned. Reading [some posts](http://www.itkeyword.com/doc/6400354897945587009/how-to-fetch-data-from-basecamp-classic-api-in-json-format ) it seems that it was not always available so not sure if it was an option at the time the project was developed.

Just tried to use projects/xxxxx/time_entries.json to report sending json in the body instead of XML and doesn't work. It seems to be a partial implementation though :(